main.cpp

```diff
@@ -336,39 +336,55 @@ int StatCompare(const CAddrReport& a, const CAddrReport& b) {
   }
 }
 
+bool isDumpDbRunning = false;
+
+extern "C" void DumpDb() {
+  if (isDumpDbRunning == true) {
+    return;
+  }
```

Comment:
Problem here is if the signal handler is invoked while a dump is in progress; it will exit DumpDb() and immediately exit the program, rather than waiting for the running dump to complete.

something like
```
if (isDumpDbRunning == true) {
    while (isDumpDbRunning) {
        sleep (1); // Wait for the dump to complete
    }
}
```
Would protect against corrupted dumps.

Reply:
Thank you for catching that @CaveSpectre11.